pkg/daemon/writer.go

```diff
@@ -121,20 +123,24 @@ func setNodeStateStatus(client snclientset.Interface, nodeName string, status sr
 }
 
 // getNodeState queries the kube apiserver to get the SriovNetworkNodeState CR
-func getNodeState(client snclientset.Interface, nodeName string) (*sriovnetworkv1.SriovNetworkNodeState, error) {
+func (w *NodeStateStatusWriter) getNodeState() (*sriovnetworkv1.SriovNetworkNodeState, error) {
 	var lastErr error
 	var n *sriovnetworkv1.SriovNetworkNodeState
 	err := wait.PollImmediate(10*time.Second, 5*time.Minute, func() (bool, error) {
-		n, lastErr = client.SriovnetworkV1().SriovNetworkNodeStates(namespace).Get(nodeName, metav1.GetOptions{})
+		n, lastErr = w.client.SriovnetworkV1().SriovNetworkNodeStates(namespace).Get(w.node, metav1.GetOptions{})
 		if lastErr == nil {
 			return true, nil
 		}
-		glog.Warningf("Failed to fetch node %s (%v); retrying...", nodeName, lastErr)
+		glog.Warningf("getNodeState(): Failed to fetch node state %s (%v); close all connections and retry...", w.node, lastErr)
+		// Use the Get() also as an client-go keepalive indicator for the TCP connection.
+		if w.OnHeartbeatFailure != nil {
```

Comment:
I think this can't be null, but if so, wouldn't it better for this to explode instead of not working? Or, if you are on the defensive side, can you add a log here?

Reply:
I think you're right.